src/averageAges.js

```diff
@@ -37,7 +47,21 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const mothersList = people.map(human => human.mother);
+
+  const mothersWithChildrens = people.filter(human =>
+    mothersList.includes(human.name)
+  );
+
+  const filterWoman = !withChildren
+    ? people.filter(human => human.sex === 'f')
+    : mothersWithChildrens;
```

Comment:
You've iterated array 3-4 times here, actually you can reduce this value to 1-2

Reply:
Just filter people array to find all woman (or all woman which names were used in `mother` properties)